src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -998,6 +998,254 @@ public with sharing class ADDR_Addresses_TEST {
+                assertEqualAddresses(listAddr[i], listCon[i]);
+                System.assertEquals(false, listAddr[i].Default_Address__c);
+            }
+        }
+    }
+
+
+    /*********************************************************************************************************
+    @description
+        Update Address' Mailing Country and State (with State & Country Picklists enabled) when the Country is empty
+        or the State empty and/or mismatched.
+    verify:
+        New Address' Country and State fields are updated with the source sObject's CountryCode and StateCode values
+    **********************************************************************************************************/
+    static testMethod void testUpdateAddressCountryAndStateWithAccountOrContactCountryCodeAndStateCode() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testUpdateAddressCountryAndStateWithAccountOrContactCountryCodeAndStateCode') {
+            return;
+        }
+
+        Integer hhSize = 1;
+        Integer contactSize = 1;
+
+        // This creates a default Address for each HH
+        createHHTestData(hhSize, contactSize);
+
+        Contact con = listConT[0];
+        con.put('MailingStateCode', 'CA');
+        con.put('MailingCountryCode', 'US');
+        con.is_Address_Override__c = true;
+
+        Address__c addressWithoutCountryState = listAddrT[0];
+        System.assertEquals('Washington', listAddrT[0].MailingState__c);
+        System.assertEquals('United States', listAddrT[0].MailingCountry__c);
+
+        // Set the Mailing State and Country to null
+        addressWithoutCountryState.MailingState__c = null;
+        addressWithoutCountryState.MailingCountry__c = null;
+        update addressWithoutCountryState;
+
+        System.assertEquals(null, addressWithoutCountryState.MailingState__c);
+        System.assertEquals(null, addressWithoutCountryState.MailingCountry__c);
+
+        Test.startTest();
+        ADDR_Addresses_TDTM.updateAddressCountryAndStateWithAccountOrContactCountryCodeAndStateCode(con, 'Mailing', addressWithoutCountryState);
+        Test.stopTest();
+
+        System.assertEquals('CA', addressWithoutCountryState.MailingState__c);
+        System.assertEquals('United States', addressWithoutCountryState.MailingCountry__c);
+    }
+
+
+    /*********************************************************************************************************
+    @description
+        Update Address' Mailing Country (with State & Country Picklists enabled) when the Country is empty.
+    verify:
+        New Address' Country field is updated with the source sObject's CountryCode value
+    **********************************************************************************************************/
+    static testMethod void testUpdateEmptyAddressMailingCountryWithAccountOrContactCountryCode() {
```

Comment:
Can the test name include expected result so anyone reviewing/modifying changes does not have to read the test to understand what is being tested? Suggestions: `testAddressCountryIsPopulatedFromCountryCodeWhenEmpty` or `testEmptyAddressCountryIsPopulatedFromCountryCode`?